internal/hardware/validator.go

```diff
@@ -130,6 +131,47 @@ func (v *validator) GetHostRequirements(role models.HostRole) models.HostRequire
 	}
 }
 
-func IsVirtSupported(inventory *models.Inventory) bool {
-	return funk.Contains(inventory.CPU.Flags, intelVirtCpuFlag) || funk.Contains(inventory.CPU.Flags, amdVirtCpuFlag)
+func (v *validator) GetClusterHostRequirements(ctx context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirements, error) {
+	operatorsRequirements, err := v.operatorsAPI.GetRequirementsBreakdownForRole(ctx, cluster, host.Role)
+	if err != nil {
+		return nil, err
+	}
+
+	hostRequirements := v.GetHostRequirements(host.Role)
+	ocpRequirements := models.ClusterHostRequirementsDetails{
+		CPUCores:                         hostRequirements.CPUCores,
+		RAMMib:                           conversions.GibToMib(hostRequirements.RAMGib),
+		DiskSizeGb:                       hostRequirements.DiskSizeGb,
+		InstallationDiskSpeedThresholdMs: hostRequirements.InstallationDiskSpeedThresholdMs,
+	}
+	total := totalizeRequirements(ocpRequirements, operatorsRequirements)
+	return &models.ClusterHostRequirements{
+		HostID:    *host.ID,
+		Ocp:       &ocpRequirements,
+		Operators: operatorsRequirements,
+		Total:     &total,
+	}, nil
+}
+
+func totalizeRequirements(ocpRequirements models.ClusterHostRequirementsDetails, operatorRequirements []*models.OperatorHostRequirements) models.ClusterHostRequirementsDetails {
+	total := models.ClusterHostRequirementsDetails{
```

Comment:
any reason why you can't reuse this variable to calculate the total and then return it, instead of copying its contents? It's a bit of a waste to throw away `ocpRequirements` in the first line :wink: .

By the way, in this case you could do a copy just by assigning the variable:
```go
total := ocpRequirements
```
https://play.golang.org/p/c-bYL60iljT

Reply:
Since the `ocpRequirements` are passed by value it is a good point:)